### PR TITLE
layernorm: FP32 SimplifiedLayerNorm/SkipSLN to prevent MoE routing collapse

### PR DESCRIPTION
**Pull request by rlegithub**
## Summary
Keeps **FP32 variance through rsqrt and the gamma multiply** for `SimplifiedLayerNormalization` / `SkipSimplifiedLayerNormalization`, so the router logits feeding the MoE stay FP32. fp16 rounding at this point collapses top-k expert routing on GPT-OSS-20B (wrong experts → garbage decode).

- `parse_simplified_layer_normalization.cpp`, `parse_skip_simplified_layer_normalization.cpp`
- `find_skip_simplified_layer_norm` isolates the SLN subgraph (`fuse_attention.cpp`) and compiles it as an MLIR group (`fuse_mlir.cpp`) preserving the FP32 gamma multiply
- `skip_simplified_layer_norm` kernel

**2 of 3 PRs** enabling GPT-OSS-20B (with the fused MoE op and GQA-attention-sink PRs).

## Test plan
- [ ] CI
- Verified on gfx1151: with FP32 SLN, GPT-OSS-20B routes correctly and decodes coherently; without it, routing collapses.

### CODE REVIEW

File: src/targets/gpu/kernels/include/migraphx/kernels/skip_simplified_layer_norm.hpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    // Each thread applies normalization in FP32 and writes fp16 output
+    for(index_int i = thread_idx; i < hidden_size; i += BLOCK_SIZE)
+    {
+        float x_val  = __half2float(input[token_idx * hidden_size + i]);
+        float sk_val = __half2float(skip[token_idx * hidden_size + i]);
+        float g_val  = __half2float(gamma[i]); // gamma converted to FP32 here
+        float result = (x_val + sk_val) * inv_std * g_val;
+        output[token_idx * hidden_size + i] = __float2half(result);
+    }
+}
```
Comment (pfultz2) [inline src/targets/gpu/kernels/include/migraphx/kernels/skip_simplified_layer_norm.hpp:106]: We already provide a layernorm kernel through the reduction kernels

File: src/targets/gpu/kernels/include/migraphx/kernels/skip_simplified_layer_norm.hpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    for(int offset = (BLOCK_SIZE / 64); offset > 0; offset >>= 1)
+        block_sum_sq += __shfl_xor(block_sum_sq, offset);
+
+    // Broadcast inv_std to all threads
+    float inv_std = __shfl(1.0f / sqrtf(block_sum_sq / (float)hidden_size + eps), 0);
+
+    // Each thread applies normalization in FP32 and writes fp16 output
+    for(index_int i = thread_idx; i < hidden_size; i += BLOCK_SIZE)
+    {
+        float x_val  = __half2float(input[token_idx * hidden_size + i]);
```
Comment (pfultz2) [inline src/targets/gpu/kernels/include/migraphx/kernels/skip_simplified_layer_norm.hpp:100]: This loads the input twice. The reduction kernel already handles this in an efficient manner,.

**Review by pfultz2 — CHANGES_REQUESTED** [review]
